test/rules/exhaustive_cases_test.dart

```diff
@@ -1,10 +1,35 @@
-// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-// test w/ `dart test -N exhaustive_cases`
+import 'package:test_reflective_loader/test_reflective_loader.dart';
 
-// Enum-like
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(ExhaustiveCasesTestLanguage219);
+    defineReflectiveTests(ExhaustiveCasesTestLanguage300);
+  });
+}
+
+abstract class BaseExhaustiveCasesTest extends LintRuleTest {
+  final actualEnumSource = r'''
+enum ActualEnum { e, f }
```

Comment:
yeah.  That's Brian's preference and while I dislike it I generally bend...  Here since I'm just migrating existing tests I'd like to keep the delta smaller.  New ones for sure.